src/transformers/models/seggpt/image_processing_seggpt.py

```diff
@@ -28,21 +28,75 @@
     PILImageResampling,
     get_channel_dimension_axis,
     infer_channel_dimension_format,
+    is_batched,
     is_scaled_image,
+    is_valid_image,
     make_list_of_images,
     to_numpy_array,
     valid_images,
 )
-from ...utils import TensorType, is_torch_available, logging, requires_backends
+from ...utils import TensorType, is_torch_available, is_vision_available, logging, requires_backends
 
 
 if is_torch_available():
     import torch
 
+if is_vision_available():
+    import PIL.Image
+
 
 logger = logging.get_logger(__name__)
 
 
+def make_list_of_masks(images: ImageInput) -> List[ImageInput]:
+    """
+    Ensure that the input is a list of images. If the input is a single image, it is converted to a list of length 1.
+    If the input is a batch of images, it is converted to a list of images. Input can be either 2D or 3D.
+
+    Args:
+        images (`ImageInput`):
+            Image of images to turn into a list of images.
+    """
+    # In the context of SegGPT we can get PIL, numpy or torch
+    # Since SegGPT can get prompt masks with zero channels (semantic maps) or 3 channels (RGB) we need to handle both
+    # The only complexity comes with torch and numpy as we have the following cases:
+    # - semantic map with 2 dimensions (H, W): torch.Tensor, numpy.ndarray
+    # - batched semantic map with 3 dimensions (B, H, W): torch.Tensor, numpy.ndarray
+    # - RGB image with 3 dimensions (C, H, W) if channel first: torch.Tensor, numpy.ndarray
+    # - batched RGB image with 4 dimensions (B, C, H, W) if channel first: torch.Tensor, numpy.ndarray
+    # If the input is then 3 dimensional there is ambiguity between batched semantic map and a single RGB image in the case where batch size is 3
+    # We can differentiate because semantic maps
+
+    if is_batched(images):
+        return images
+
+    # Either the input is a single image, in which case we create a list of length 1
+    if isinstance(images, PIL.Image.Image):
+        # PIL images are never batched
+        return [images]
+
+    if is_valid_image(images):
+        if images.ndim == 2:
+            # Singe semantic map
+            images = [images]
+        elif images.ndim == 3:
+            # Either single RGB or batched semantic map
+            # Assuming that if first or last dimension is 3 (Channel first or last) it is RGB
+            images = [images] if (images.shape[0] == 3 or images.shape[-1] == 3) else list(images)
```

Comment:
Indeed not ideal, therefore I made a new proposal here https://github.com/huggingface/transformers/pull/29550/commits/f372a40a4fce61e85ec0ce61e4a8a63e445b8e6a

Instead of having to deal with that, I added a new optional argument called `segmentation_maps` that follows other image processor's conventions. So, to obtain the `prompt_masks` in the `BatchedFeature` one should specify `prompt_masks` or `segmentation_maps` where `segmentation_maps` should be the 2-dimensional input that we convert to RGB when `num_labels` is specified.